### PR TITLE
feat(web-app): add developing page

### LINKED ISSUE #779: Create new page - Developing
Left nav reference https://github.com/carbon-design-system/carbon-platform/issues/439

Figma file  https://www.figma.com/file/uQXFGSKSJRr0l8CDd3cRGw/v1-prototype?node-id=0%3A1

Screen Shot 2022-05-25 at 9 27 22 AM

### PR DESCRIPTION
**Pull request by alisonjoseph**
Closes #779

Add Developing page

#### Changelog

**New**

- Add icons to icon packages
  - js
  - npm
  - svelte
  - vue
- Add icons to mdx icon component
  - js
  - npm
  - svelte
  - vue
  - codepen

**Removed**

- {{ Removed thing }}

### CODE REVIEW

File: services/web-app/pages/developing/index.mdx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  </MiniCard>
+  <MiniCard title="Vue tutorial" href="/assets/carbon-components-vue">
+    <MdxIcon name="vue" />
+  </MiniCard>
+  <MiniCard title="Svelte tutorial" href="h/assets/carbon-components-svelte">
+    <MdxIcon name="svelte" />
+  </MiniCard>
+  <MiniCard title="Vanilla tutorial" href="/assets/carbon-components">
+    <MdxIcon name="js" />
+  </MiniCard>
```
Comment (francinelucca) [inline services/web-app/pages/developing/index.mdx:60]: these ones are not showing up for me 
image

File: services/web-app/pages/developing/index.mdx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+drives IBM’s design philosophy and principles, and can make informed decisions in your product work.
+
+By adopting Carbon, front-end developers and engineers gain a collection of reusable components to
+build websites and user interfaces quickly and efficiently.
+
+### Understand the ecosystem
+
+The following diagram shows the relationships of assets and guidance across IBM. The Carbon Design
+System is often extended or complimented by additional product and digital systems.
+
```
Comment (francinelucca) [inline services/web-app/pages/developing/index.mdx:30]: are we missing a picture here? 
image

File: services/web-app/pages/developing/index.mdx [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+## Step 2: Get Started
```
Comment (francinelucca) [inline services/web-app/pages/developing/index.mdx:31]: is this the correct spacing (h2 to h3)? looks a bit different to me but couldn't find exact measurements on the figma page 🤔 
image
image

**Review by francinelucca — CHANGES_REQUESTED** [review]

File: services/web-app/pages/developing/index.mdx (hunk 3 above)
Comment (alisonjoseph) [reply services/web-app/pages/developing/index.mdx:33]: The spacing matches the current website / Gatsby theme, so I think we're good. I think the sketch file is off. OR if design wants a global change we can make a new issue to adjust any spacing site-wide.

**Review by francinelucca — DISMISSED** [review]
💃

File: services/web-app/pages/developing/index.mdx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    title="Vue app tutorial"
+    href="https://carbondesignsystem.com/developing/vue-tutorial/overview/"
+  >
+    <MdxIcon name="vue" />
+  </MiniCard>
+</CardGroup>
+
+### Explore additional libraries
+
+Product and digital teams at IBM have additional Libraries that are built upon the Carbon Design
```
Comment (andreancardona) [inline services/web-app/pages/developing/index.mdx:94]:
```suggestion
Product and digital teams at IBM have additional [Libraries](https://next.carbondesignsystem.com/assets/libraries) that are built upon the Carbon Design
```

File: services/web-app/pages/developing/index.mdx [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+## Step 2: Get Started
+
+### Choose your framework
+
+The Carbon Design System is built React first, with official support for Web Components. We also
+support core parts of the system in vanilla JS, Angular, Vue, and Svelte—and we offer guidelines for
+other frameworks.
+
+Head over to [Libraries](/assets/libaries) to complete the installation tutorial for your framework
```
Comment (andreancardona) [inline services/web-app/pages/developing/index.mdx:41]:
```suggestion
Head over to [Libraries](/assets/libraries) to complete the installation tutorial for your framework
```

**Review by andreancardona — CHANGES_REQUESTED** [review]
2 super small changes but otherwise looks good 🔥

**Comment by alisonjoseph** [comment]
Good catch @andreancardona!

**Review by andreancardona — APPROVED** [review]

**Review by francinelucca — APPROVED** [review]
LGTM!

**Pull request merged**